### PR TITLE
Abstract data actions

### PR DESCRIPTION
**Pull request by davight**
This Pull-Request adds support for ObjectTag to define its own data actions. 
(Keeping this as draft for now to hear your opinions on the design and everything)

Related discussion thread: https://discord.com/channels/315163488085475337/1468638851802075270

# Additions:
- New interface `Actionable<T extends ObjectTag>` that provides 4 methods `xOperation(ObjectTag value, TagContext context)`, where `x` is `substraction`, `multiplication`, `division`, `addition`. With default implementation of throwing exception that the given object does not support given operation.
- DurationTag support for addition and subtraction operation. (`DurationTag:+:DurationTag`, `DurationTag:-:DurationTag`)
- TimeTag support for addition and subtraction operation. (`TimeTag:+:DurationTag`, `TimeTag:-:DurationTag`)
- VectorTag support for addition, subtraction, multiplication and division (`VectorTag:+:VectorTag`, `VectorTag:-:VectorTag`, `VectorTag:*:ElementTag`, `VectorTag:/:ElementTag`)
- Support for accessing first value of ListTag (in DataActionHelper) using `first` just like we have for last using `last`

# Changes:
- Moved ElementTag support for addition, subtraction, multiplication and division to the `ElementTag.class` (for the new Actionable system)
- Cleaned up the `DataAction.class`
- Minor cleanup of `DataActionHelper.class`

### CODE REVIEW

**Comment by mcmonkey4eva** [comment]
Please run before/after tests on common cases and edge cases for basic numeric element handling. How is invalid data (eg add/mul/div/whatever on a flag but the flag's value is not a number) handled? How is initial value handled (`- flag server myval:+:1` when the flag doesn't exist yet)? etc.
All these tests should be the same before & after, unless there's an intentional deviation. Also test to make sure that the handling of the different odd cases is consistent between data types and makes sense (what happens if you have a nonexistent flag and `- flag server myval:+:5s`?)

File: src/main/java/com/denizenscript/denizencore/utilities/data/DataAction.java [hunk 1]
```diff
@@ -35,141 +35,37 @@ public String toString() {
...
-                break;
+                return;
             }
-            case REMOVE: {
+            case REMOVE -> {
                 ListTag list = autoList(key, context);
                 if (index != 0) {
                     if (index == Integer.MAX_VALUE && !list.isEmpty()) {
-                        list.remove(list.size() - 1);
+                        list.removeLast();
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/denizencore/utilities/data/DataAction.java]: I think this is a jdk version break
